src/cmd/manage_nsfs.js

```diff
@@ -351,12 +351,26 @@ async function fetch_account_data(action, user_input) {
 
     // override values
     if (has_access_keys(data.access_keys)) {
-        // access_key as SensitiveString
-        data.access_keys[0].access_key = _.isUndefined(data.access_keys[0].access_key) ? undefined :
-            new SensitiveString(String(data.access_keys[0].access_key));
-        // secret_key as SensitiveString
-        data.access_keys[0].secret_key = _.isUndefined(data.access_keys[0].secret_key) ? undefined :
-            new SensitiveString(String(data.access_keys[0].secret_key));
+        for (const access_key_object of data.access_keys) {
+            // access_key as SensitiveString
+            if (access_key_object.access_key) {
+                access_key_object.access_key = new SensitiveString(String(access_key_object.access_key));
+            } else {
+                // since the _.OmitBy only works for the first level of properties,
+                // and we might get here with data.access_keys of [ { access_key: undefined, secret_key: undefined } ]
+                // from the function get_access_keys
+                delete access_key_object.access_key;
+            }
+            if (access_key_object.secret_key) {
+                access_key_object.secret_key = new SensitiveString(String(access_key_object.secret_key));
+            } else {
+                // since the _.OmitBy only works for the first level of properties,
+                // and we might get here with data.access_keys of [ { access_key: undefined, secret_key: undefined } ]
+                // from the function get_access_keys - the access_key will be replaced by the user access_key, but
```

Comment:
I was rethinking it, and I will not add this commit to this PR it will be a gap that I can solve in a different PR (and probably differently).

Anyway, this is the explanation after editing it:
```js
// since the _.OmitBy only works for the first level of properties,
// and we might get here with data.access_keys of [ { access_key: undefined, secret_key: undefined } ]
// from the function get_access_keys - the access_key with value undefined will be replaced by the user access_key,
// but the secret_key with value undefined won't be replaced (we will have the encrypted_secret_key) in case of ACTIONS.DELETE
// since the access keys are not decrypted
```

I will rebase and merge only the commit that was approved.

Reply:
Not included in this PR, mentioned as a gap.